src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacPropertyAttributes/DacPropertyAttributesAnalyzer.cs

```diff
@@ -67,28 +67,28 @@ private static Task AnalyzePropertyAsync(SymbolAnalysisContext symbolContext, PX
 			if (!validSpecialTypes)
 				return;
 
-			CheckForPXDBCalcedAndUnboundTypeAttributes(symbolContext, pxContext, fieldAttributesRegister, property);
+			CheckForPXDBCalcedAndUnboundTypeAttributes(symbolContext, pxContext, fieldAttributesRegister, property, attributesWithInfos);
 
 			await CheckForFieldTypeAttributesAsync(property, symbolContext, pxContext, attributesWithInfos)
 					.ConfigureAwait(false);
 		}
 
 		private static void CheckForPXDBCalcedAndUnboundTypeAttributes(SymbolAnalysisContext symbolContext, PXContext pxContext,
-			FieldTypeAttributesRegister fieldAttributesRegister, IPropertySymbol propertySymbol)
+			FieldTypeAttributesRegister fieldAttributesRegister, IPropertySymbol propertySymbol,
+			List<(AttributeData Attribute, List<FieldTypeAttributeInfo> Infos)> attributesWithInfos)
 		{
 			symbolContext.CancellationToken.ThrowIfCancellationRequested();
 
-			var attributes = propertySymbol.GetAttributes();
 			var hasUnboundTypeAttribute = false;
 			var hasPXDBCalcedAttribute = false;
 
-			foreach (var attr in attributes)
+			foreach (var (attribute, info) in attributesWithInfos)
 			{
-				if (attr.AttributeClass.Equals(pxContext.FieldAttributes.PXDBCalcedAttribute))
+				if (attribute.AttributeClass.Equals(pxContext.FieldAttributes.PXDBCalcedAttribute))
```

Comment:
@metallmonster this validation is not valid. The `attribute` is the same attribute data that you received from the property.
You should not validate it directly, instead you need to use `info` DTO. There is `Kind` property there which will help to find out if the attribute is PXDBCalced. 
To find out if the attribute is field type attribute and is DB unbound you would have to either modify DTO to add this info and `FieldTypeAttributesRegister`

Reply:
@SENya1990 I agree with the first part of your comment. I'm already using `FieldTypeAttributesRegister` to check whether an attribute is an unbound type or not

Reply:
@SENya1990 I have looked into the source code and have understood your idea.